kernel/src/filesystem/fat/entry.rs

```diff
@@ -2038,9 +2035,10 @@ impl ShortNameGenerator {
                 .map(|s| u16::from_str_radix(s, 16));
             // 如果校验码相同
             if checksum_result == Ok(Ok(self.checksum)) {
-                let num = num_suffix.unwrap();
                 // 置位checksum_bitmask中，基础名末尾数字的对应位
-                self.checksum_bitmask |= 1 << num;
+                if let Some(num) = num_suffix {
+                    self.checksum_bitmask |= 1 << num;
+                }
```

Comment:
应该保有分支设置kwarn!()标识而不是当作什么都没发生？
```suggestion
                if let Some(num) = num_suffix {
                    self.checksum_bitmask |= 1 << num;
                } else {
                    kwarn!("num_suffix shouldn't be None!");
                }
```
或直接
```suggestion
                self.checksum_bitmask |= 1 << num_suffix.unwrap();
```

Reply:
fat标准规定了，只有基础名冲突的时候才会有后缀。